src/operators/expression.md

```diff
@@ -0,0 +1,121 @@
+| `expression`        | `Expression`                                     | Expression script                                                                                           | <pre><code>(A - B) / (A + B)</code></pre>                                                                |
+| `outputType`        | [`RasterDataType`](/datatypes/rasterdatatype.md) | A raster data type for the output                                                                           | <pre><code>U8</code></pre>                                                                               |
+| `outputNoDataValue` | `Number`                                         | NO DATA value for the output                                                                                | <pre><code>-2</code></pre>                                                                               |
+| `outputMeasurement` | [`Measurement`](/datatypes/measurement.md)       | Description about the output                                                                                | <pre><code>{<br>&nbsp;&nbsp;"type": "continuous",<br>&nbsp;&nbsp;"measurement": "NDVI"<br>}</code></pre> |
+| `mapNoData`         | `Boolean`                                        | Should NO DATA values be mapped with the `expression`? Otherwise, they are mapped automatically to NO DATA. | <pre><code>false</code></pre>                                                                            |
+
+## Types
+
+The following describes the types used in the parameters.
+
+### Expression
+
+Expressions are simple scripts to perform pixel-wise computations.
+One can refer to the raster inputs as `A` for the first raster, `B` for the second, and so on.
+Furthermore, expressions can use the variables `is_A_nodata`, `is_A_nodata`, etc. to check for NO DATA values.
+This is important if `mapNoData` is set to true.
+Otherwise, NO DATA values are mapped automatically to the output NO DATA value.
+Finally, the value `out_nodata` can be used to output NO DATA.
+
+Users can basically think of this implicit function signature for, e.g., two inputs:
+
+```Rust
+fn (A: f64, is_A_nodata: bool, B: f64, is_B_nodata: bool, out_nodata: f64) -> f64
+```
+
+As a start, expressions contain algebraic operations and mathematical functions.
+
+```Rust
+(A + B) / 2
+```
+
+In addition, branches can be used to check for conditions.
+
+```Rust
+if is_A_nodata {
+    B
+} else {
+    A
+}
+```
+
+Function calls can be used to access utility functions.
+
+```Rust
+max(A, 0)
+```
+
+To generate more complex expressions, it is possible to have variable assignments.
+
+```Rust
+let mean = (A + B) / 2;
+let coefficient = 0.357;
+mean * coefficient
+```
+
+Note, that all assignments are separated by semicolons.
+However, the last expression must be without semicolon.
```

Comment:
maybe it would make sense to also support a `return` keyword. I guess none of our users will know rust so in general I am not convinced that it is a good idea to model the expression language like rust. It also sounds kind of arbitrary to omit the semicolons if one does not know the difference between statements and expressions...

Reply:
Writing `A + B` as an expression comes naturally, so omitting a `return` seems to be the most natural thing to do.
I guess users just need to know what to do if they use variables.